backend/packages/Upgrade/src/api/services/ExperimentAssignmentService.ts

```diff
@@ -458,19 +458,19 @@ export class ExperimentAssignmentService {
 
     // query all experiment and sub experiment
     // check if user or group is excluded
-    let experiments: Experiment[] = [],
-      userExcluded: string,
-      groupExcluded: string[];
+    let experiments: Experiment[] = [];
+
+    const [userExcluded, groupExcluded] = await this.checkUserOrGroupIsGloballyExcluded(experimentUser);
+
+    if (userExcluded || groupExcluded.length > 0) {
```

Comment:
This change to exclude before fetching the experiments makes sense, but it was never mentioned that it would be part of this hotfix as far as I can recall. It's probably fine, but please mention all changes in the comments, this is a significant change to test for that I don't recall discussing making within the hotfix branch.

Reply:
You are correct, this was not discussed and we found this improvement while discussion an issue in our scrum meet and thought of rectifying this in the current hotfix, as the same code section was being updated. We will keep this in mind for future. I will add this in PR description.